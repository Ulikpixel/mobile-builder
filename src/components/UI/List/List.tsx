import React, { FC } from 'react'
import { FlatList, Text } from 'react-native'
import styled from 'styled-components/native'
import { ITheme } from '../../../types/theme'

interface ListProps {
  data: {
    key: string
  }[]
}

export const ListIndex = styled.Text<ITheme>`
  color: ${props => props.theme.colors['organick-dark']};
  font-size: 14px;
  line-height: 20px;
`

ListIndex.defaultProps = {
  style: {
    fontFamily: 'open-sans-bold',
  },
}

export const ListItem = styled.View`
  display: flex;
  gap: 3px;
  flex-direction: row;
  margin-bottom: 10px;
  align-items: center;
`

const List: FC<ListProps> = ({ data }) => (
  <FlatList
    data={data}
    renderItem={({ item, index }) => (
      <ListItem>
        <p>hello</p>
        <ListIndex>{index}.</ListIndex>
        <Text>{item.key}</Text>
      </ListItem>
    )}
  />
)

export default List
