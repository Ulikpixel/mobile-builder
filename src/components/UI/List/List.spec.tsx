import * as React from 'react'
import renderer from 'react-test-renderer'
import List from './List'

it(`renders correctly`, () => {
  const exampleList = ['one', 'two']
  const tree = renderer.create(<List data={exampleList} />).toJSON()

  expect(tree).toMatchSnapshot()
})
