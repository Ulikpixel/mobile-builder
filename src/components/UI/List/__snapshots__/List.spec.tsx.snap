// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    [
      {
        "key": "one",
      },
      {
        "key": "two",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "gap": 3,
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#525C60",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "open-sans-bold",
              },
            ]
          }
        >
          0
          .
        </Text>
        <Text>
          one
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "gap": 3,
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#525C60",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "open-sans-bold",
              },
            ]
          }
        >
          1
          .
        </Text>
        <Text>
          two
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
